File: elixir/atbash-cipher/atbash.exs
defmodule Atbash do

  @atbash %{
    "a" => "z", "b" => "y", "c" => "x", "d" => "w", "e" => "v", "f" => "u",
    "g" => "t", "h" => "s", "i" => "r", "j" => "q", "k" => "p", "l" => "o",
    "m" => "n", "n" => "m", "o" => "l", "p" => "k", "q" => "j", "r" => "i",
    "s" => "h", "t" => "g", "u" => "f", "v" => "e", "w" => "d", "x" => "c",
    "y" => "b", "z" => "a"
  }

  @doc """
  Encode a given plaintext to the corresponding ciphertext

  ## Examples

  iex> Atbash.encode("completely insecure")
  "xlnko vgvob rmhvx fiv"
  """
  @spec encode(String.t) :: String.t
  def encode(plaintext) do
    plaintext
    |> String.downcase
    |> String.replace(~r{[^a-z0-9]}, "")
    |> String.graphemes
    |> Enum.map(&(@atbash[&1] || &1))
    |> split_into_groups_of(5, [])
    |> Enum.join(" ")
  end

  defp split_into_groups_of([], _, acc), do: Enum.reverse(acc)
  defp split_into_groups_of(strs, n, acc) do
    split_into_groups_of(Enum.drop(strs, n), n,
                         [Enum.take(strs, n) |> Enum.join("") | acc])
  end
end

File: go/atbash-cipher/atbash.go
package atbash

import (
	"strings"
	"unicode"
)

const splitLen = 5

var atbash = map[rune]rune{
	'a': 'z', 'b': 'y', 'c': 'x', 'd': 'w', 'e': 'v', 'f': 'u',
	'g': 't', 'h': 's', 'i': 'r', 'j': 'q', 'k': 'p', 'l': 'o',
	'm': 'n', 'n': 'm', 'o': 'l', 'p': 'k', 'q': 'j', 'r': 'i',
	's': 'h', 't': 'g', 'u': 'f', 'v': 'e', 'w': 'd', 'x': 'c',
	'y': 'b', 'z': 'a',
}

func Atbash(plaintext string) string {
	s := []rune{}
	for _, c := range strings.ToLower(plaintext) {
		atbash := atbash[c]
		if atbash != 0 {
			s = append(s, atbash)
		} else if unicode.IsNumber(c) {
			s = append(s, c)
		}
	}
	return strings.Join(splitIntoGroups(string(s)), " ")
}

func splitIntoGroups(s string) (strs []string) {
	for len(s) > 0 {
		if splitLen > len(s) {
			strs = append(strs, s)
			s = ""
		} else {
			strs = append(strs, s[0:splitLen])
			s = s[splitLen:]
		}
	}
	return
}
